### DISCUSSION TITLE
Handle failure modes

### DISCUSSION THREAD

**Discussion by haansn08** [spacecowboy/Feeder#257] [Ideas]
I've been using Feeder for a while now, so I have been wondering how I can recognize if RSS feeds become unavailable or are moved to a different location. I can think of following scenarios:
* We receive a [`301 Moved Permanently`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/301) status code: In that case the URL of the feed should probably be updated without user interaction.
* We receive a [`404 Not Found`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404): The feed is likely dead and the user should be prompted if they want to remove the feed from their list.
* The domain does not exist anymore: Should probably be handled the same way as a HTTP 404.

Currently, it seems such failure modes are silently dropped and I have little confidence I could recognize if a feed stops existing or is moved somewhere else.

**Comment by DamonHD** [comment]
There is also 308 code?

Also link in podcast feed.

Also itunes:new-feed-url tag in podcast feed.

Rgds

Damon

**Comment by spacecowboy** [comment]
Feeder used to auto update the URL but this is a bad idea, in case of password protected feeds.

I'd want to store the error messages like AntennaPod

![Screenshot_20240226_120448_AntennaPod](https://github.com/spacecowboy/Feeder/assets/223655/b839659a-21c4-4f79-92e7-f1ab956954a8)

**Reply by haansn08** [reply]
>  in case of password protected feeds.

Do password protected feeds usually answer with HTTP redirects to other URLs?

> I'd want to store the error messages like AntennaPod

If possible, there shouldn't be a popup for every error which might be annoying when there are multiple errors at the same time. Maybe a red error card instead of an article card?

**Comment by spacecowboy** [comment]
> > in case of password protected feeds.
> 
> Do password protected feeds usually answer with HTTP redirects to other URLs?

Yes because I got a bug report about just that.

One example is if a server (incorrectly!) supports both http and https, and one fixes this. They then do 301s for the HTTP requested feeds, but crucially (what feeder cares about), is that they put the "self-link" inside the feed and point to the new HTTPS feed.

But, since it's password protected, the URL in the feed itself is without any authentication details.

Showing a yellow warning flag on the feed would be appropriate in this case I think.

> If possible, there shouldn't be a popup for every error which might be annoying when there are multiple errors at the same time. Maybe a red error card instead of an article card?

Oh maybe that's not clear. AntennaPod only displays information on the feeds themselves. You only get the pop up if you click to get more details (Look at the red thing behind the popup, that's the button).
